lib/datadog/core/transport/http/adapters/net.rb

```diff
@@ -16,14 +16,11 @@ class Net
               :timeout,
               :ssl
 
-            # in seconds
-            DEFAULT_TIMEOUT = 30
-
             # @deprecated Positional parameters are deprecated. Use named parameters instead.
             def initialize(hostname = nil, port = nil, **options)
               @hostname = hostname || options.fetch(:hostname)
               @port = port || options.fetch(:port)
-              @timeout = options[:timeout] || DEFAULT_TIMEOUT
+              @timeout = options[:timeout] || Datadog::Core::Transport::Ext::HTTP::DEFAULT_TIMEOUT_SECONDS
               @ssl = options.key?(:ssl) ? options[:ssl] == true : false
             end
 
```

Comment:
Looking at the code, I don't think anyone is calling this `initialize` anymore -- it always gets called by `build`. Being that the case, my suggestion would be to actually simplify initialize to directly receive the `agent_settings`.

It's a bit less code + no need to access the timeout constant, it always gets read from the agent settings resolver.